### PR TITLE
Add table oci_objectstorage_object. Closes #338

### LINKED ISSUE #338: Add table oci_objectstorage_bucket_objects
https://registry.terraform.io/providers/hashicorp/oci/latest/docs/resources/objectstorage_object
https://docs.oracle.com/en-us/iaas/Content/Object/Concepts/objectstorageoverview.htm

### PR DESCRIPTION
**Pull request by c0d3r-arnab**
# Integration test logs
**Tests not feasible**

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  name,
  bucket_name,
  namespace,
  region,
  size,
  md5,
  time_created,
  time_modified
from
  oci_objectstorage_object;
+-------------------------+----------------------+--------------+--------------+--------+--------------------------+---------------------------+---------------------------+
| name                    | bucket_name          | namespace    | region       | size   | md5                      | time_created              | time_modified             |
+-------------------------+----------------------+--------------+--------------+--------+--------------------------+---------------------------+---------------------------+
| test-obj-1object.txt    | test-buket           | bmqeqvslavsz | ap-mumbai-1  | 517    | Q9MZB7Gof0gwLmQxy7eC8g== | 2022-01-06T14:21:38+05:30 | 2022-01-06T14:21:38+05:30 |
| test-obj-rootoutput.txt | bucket-20210809-1006 | bmqeqvslavsz | ap-mumbai-1  | 301627 | sJEN1kBLnBYNd2yrsa9+Og== | 2022-01-17T12:42:31+05:30 | 2022-01-17T12:42:31+05:30 |
| test-obj-1object.txt    | test-buket           | bmqeqvslavsz | us-ashburn-1 | 517    | Q9MZB7Gof0gwLmQxy7eC8g== | 2022-01-06T14:24:09+05:30 | 2022-01-06T14:24:09+05:30 |
| test-obj-2object-2.txt  | test-bucket-archive  | bmqeqvslavsz | us-ashburn-1 | 33     | ylXBDBbwMN/612uBh+xKZg== | 2022-01-12T12:45:14+05:30 | 2022-01-12T12:45:14+05:30 |
| test-obj-1object.txt    | test-bucket-archive  | bmqeqvslavsz | us-ashburn-1 | 33     | /0XMODUWUwfvQUwjyixvZw== | 2022-01-12T12:44:56+05:30 | 2022-01-12T12:44:56+05:30 |
| test-obj-2object-2.txt  | test-buket           | bmqeqvslavsz | us-ashburn-1 | 33     | ylXBDBbwMN/612uBh+xKZg== | 2022-01-12T12:58:53+05:30 | 2022-01-12T12:58:53+05:30 |
+-------------------------+----------------------+--------------+--------------+--------+--------------------------+---------------------------+---------------------------+
> select
  name,
  bucket_name,
  namespace,
  region,
  archival_state
from
  oci_objectstorage_object
where
  archival_state = 'Archived';
+------------------------+---------------------+--------------+--------------+----------------+
| name                   | bucket_name         | namespace    | region       | archival_state |
+------------------------+---------------------+--------------+--------------+----------------+
| test-obj-1object.txt   | test-bucket-archive | bmqeqvslavsz | us-ashburn-1 | Archived       |
| test-obj-2object-2.txt | test-bucket-archive | bmqeqvslavsz | us-ashburn-1 | Archived       |
+------------------------+---------------------+--------------+--------------+----------------+
> select * from oci_objectstorage_object limit 1
+----------------------+-------------+--------------+----------------+---------------+---------------------+------------------+------------------+---------------+--------------+---------------------------
| name                 | bucket_name | namespace    | archival_state | cache_control | content_disposition | content_encoding | content_language | content_range | content_type | etag                      
+----------------------+-------------+--------------+----------------+---------------+---------------------+------------------+------------------+---------------+--------------+---------------------------
| test-obj-1object.txt | test-buket  | bmqeqvslavsz |                | <null>        | attachment          | <null>           | <null>           | <null>        | text/plain   | d558915e-813e-461e-b506-e0
+----------------------+-------------+--------------+----------------+---------------+---------------------+------------------+------------------+---------------+--------------+---------------------------
```
</details>

### CODE REVIEW

File: oci/table_oci_objectstorage_object.go [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+				Hydrate:     getObjectStorageObject,
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "etag",
+				Description: "The current entity tag (ETag) for the object.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "expires",
```
Comment (bigdatasourav) [inline oci/table_oci_objectstorage_object.go:89]: The column name is incorrect, not matching with the from the field.

File: oci/table_oci_objectstorage_object.go [hunk 2]
```diff
@@ -0,0 +1,283 @@
...
+				Description: "The date and time the object was modified.",
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeModified.Time"),
+			},
+			{
+				Name:        "time_of_archival",
+				Description: "Time that the object is returned to the archived state. This field is only present for restored objects.",
+				Hydrate:     getObjectStorageObject,
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("TimeCreated.Time"),
```
Comment (bigdatasourav) [inline oci/table_oci_objectstorage_object.go:140]: From field value is incorrect.

File: oci/table_oci_objectstorage_object.go [hunk 3]
```diff
@@ -0,0 +1,283 @@
...
+func tableObjectStorageObject(_ context.Context) *plugin.Table {
+	return &plugin.Table{
+		Name:        "oci_objectstorage_object",
+		Description: "OCI Object Storage Object",
+		// Object can have same name in two different buckets, regions or compartments, leading to duplicate result in get call
+		List: &plugin.ListConfig{
+			Hydrate:       listObjectStorageObjects,
+			ParentHydrate: listObjectStorageBuckets,
+		},
+		GetMatrixItem: BuildCompartementRegionList,
```
Comment (bigdatasourav) [inline oci/table_oci_objectstorage_object.go:25]: Can we use BuildRegionList as we are not using the Compartment?

**Review by bigdatasourav — APPROVED** [review]
LGTM

File: docs/tables/oci_objectstorage_object.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# Table: oci_objectstorage_object
+
+In the Oracle Cloud Infrastructure Object Storage service, an object is a file or unstructured data you upload to a bucket within a compartment  within an Object Storage namespace. The object can be any type of data, for example, multimedia files, data backups, static web content, or logs.
```
Comment (misraved) [inline docs/tables/oci_objectstorage_object.md:3]:
```suggestion
In the Oracle Cloud Infrastructure Object Storage service, an object is a file or unstructured data you upload to a bucket within a compartment within an Object Storage namespace. The object can be any type of data, for example, multimedia files, data backups, static web content, or logs.
```

**Comment by rajlearner17** [comment]
@c0d3r-arnab example query results below
```
Error: rpc error: code = Internal desc = hydrate function listObjectStorageBuckets failed with panic interface conversion: interface {} is nil, not string (SQLSTATE HV000)```

File: oci/table_oci_objectstorage_object.go [hunk 5]
```diff
@@ -0,0 +1,283 @@
...
+			{
+				Name:        "expires",
+				Description: "The date and time after which the object is no longer cached by a browser, proxy, or other caching entity.",
+				Hydrate:     getObjectStorageObject,
+				Type:        proto.ColumnType_TIMESTAMP,
+				Transform:   transform.FromField("Expires.Time"),
+			},
+			{
+				Name:        "is_not_modified",
+				Description: "Flag to indicate whether or not the object was modified.  If this is true,the getter for the object itself will return null.",
```
Comment (misraved) [inline oci/table_oci_objectstorage_object.go:97]:
```suggestion
				Description: "Flag to indicate whether or not the object was modified. If this is true, the getter for the object itself will return null.",
```

File: oci/table_oci_objectstorage_object.go [hunk 6]
```diff
@@ -0,0 +1,283 @@
...
+				Hydrate:     getObjectStorageObject,
+				Type:        proto.ColumnType_BOOL,
+				Default:     false,
+			},
+			{
+				Name:        "md5",
+				Description: "Base64-encoded MD5 hash of the object data.",
```
Comment (misraved) [inline oci/table_oci_objectstorage_object.go:104]: Does the `md5` column include sensitive data?

**Review by misraved — COMMENTED** [review]
Make changes to the description of a few suggested columns

File: oci/table_oci_objectstorage_object.go (hunk 6 above)
Comment (c0d3r-arnab) [reply oci/table_oci_objectstorage_object.go:104]: This does not contain any sensitive data.

**Review by rajlearner17 — APPROVED** [review]
Looks good after testing multiple queries and formats.

**Pull request merged**